test/exporter-sharded.spec.js

```diff
@@ -150,4 +182,267 @@ describe('exporter sharded', function () {
       }
     ], done)
   })
+
+  it('exports all files from a sharded directory with subshards', (done) => {
+    waterfall([
+      (cb) => createShard(31, cb),
+      (dir, cb) => {
+        pull(
+          exporter(`/ipfs/${dir.toBaseEncodedString()}`, ipld),
+          pull.collect(cb)
+        )
+      },
+      (exported, cb) => {
+        try {
+          expect(exported.length).to.equal(32)
+
+          const dir = exported.shift()
+
+          expect(dir.type).to.equal('dir')
+
+          exported.forEach(file => expect(file.type).to.equal('file'))
+
+          cb()
```

Comment:
Don't call callbacks inside try/catch blocks

Reply:
At the time I found it made the tests fail faster. Of course now it doesn't seem to make any difference, have removed them all.